core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/SimplifyExpressions.java

```diff
@@ -49,6 +50,7 @@ public static Expression rewrite(Expression expression, Session session, SymbolA
         expression = pushDownNegations(plannerContext.getMetadata(), expression, expressionTypes);
         expression = extractCommonPredicates(plannerContext.getMetadata(), expression);
         expression = normalizeOrExpression(expression);
+        expression = normalizeOrInExpression(expression);
```

Comment:
Is it possible to merge with `normalizeOrExpression` ? Are we planning to do as a followup ?

Reply:
Addressed here - https://github.com/trinodb/trino/pull/17635/files#r1206161868